src/daft-plan/src/sink_info.rs

```diff
@@ -50,7 +50,11 @@ pub enum CatalogType {
 pub struct IcebergCatalogInfo {
     pub table_name: String,
     pub table_location: String,
-    pub spec_id: i64,
```

Comment:
Can we keep `spec_id` if possible? Will be cleaner to not add more cruft onto the plans and add more serde overhead, especially when it comes to PyObjects like this.

Reply:
I don't think we can, since the file writer itself does not have access to the iceberg table metadata. We could extract out the data from the partitioning spec and put it into a struct but we would need to either recreate it in the writer or reimplement a lot of the methods that pyiceberg provides.